kedro/io/data_catalog.py

```diff
@@ -311,6 +364,28 @@ def _get_dataset(
 
         return data_set
 
+    def _resolve_dataset(
```

Comment:
I think this definitely conflicts with `_suggest_resume_scenario`, can you try to test with a fail pipeline and see if it breaks it?

Reply:
fixed it

Reply:
I checked, it doesn't break `_suggest_resume_scenario`